app/controllers/carts_controller.rb

```diff
@@ -11,17 +11,35 @@ def add_item
     redirect_back(fallback_location: restaurants_path)
   end
 
-  def checkout
-    @order = ShoppingCart.find(params[:format])
-    session.delete(:cart_id)
-    flash[:notice] = 'Your food is on its way!'
-    # In a later feature this needs to create some action item to actually make the order happen.
-    # Restrict this function to a Customer
+  def create
+    link_user_and_cart(params)
+    charge = StripePayment.perform_payment(params, @cart)
+    @order = ShoppingCart.find_by(user_id: current_user.id)
+    if charge.class == Stripe::Charge
+      mark_cart_as_paid(charge)
+      session.delete(:cart_id)
+      flash[:notice] = 'Your food is on its way!'
+      render :checkout
+    else
+      redirect_to carts_path, alert: charge
```

Comment:
ok very well. So... should we get Raoul to look at this or is one of us going to do the dogfood thing?